[javanaut][operationExpr][1/2]feat: Add Assignment operation expr with multiply and assignment operator

summer-ji-eng opened the pull request:
enables *= operator.
^= will come with next PR [2/2]

type-checking tables in go/javanaut-assignment-operation-expr

miraleung reviewed (changes requested):
PTAL at the comments in the design doc.

summer-ji-eng commented:
Addressed the comments in the design doc with changes:
-  Update the convenience function name using your suggestion.
-  Rename lhsExpr to variableExpr and rhsExpr to valueExpr. Add check to make sure variable expr is not decl with unit tests.
- Add comments.
